src/routes/(auth)/sign-up/+page.server.js

```diff
@@ -1,29 +1,41 @@
-import { z } from "zod";
-
-const signupSchema = z.object({
-	name: z
-		.string({ required_error: "name is required" })
-		.trim()
-		.min(2, { message: "name is too short" }),
-	email: z.string({ required_error: "email is required" }).email(),
-	password: z
-		.string({ required_error: "password is required" })
-		.min(4, { message: "password is too short" }),
-});
+import { fail, redirect } from '@sveltejs/kit';
+import { signupSchema } from '$lib/server/validator/authSchema.js';
 
 /** @type {import('./$types').Actions} */
 export const actions = {
-	default: async ({ request }) => {
-		const form = Object.fromEntries(await request.formData());
+  default: async ({ request, fetch }) => {
+    const form = Object.fromEntries(await request.formData());
+
+    const { data, error: validationError, success } = signupSchema.safeParse(form);
+
+    if (!success) {
+      const errors = validationError.flatten().fieldErrors;
+      const firstError = Object.values(errors).flat().at(0);
+      return fail(400, { error: firstError });
+    }
+
+    try {
+      const { email, password, name } = data;
```

Comment:
I see no need to destructure this, since the `data` contains the `name`, `email`, `password` only.

Reply:
removed the destructure for the sign-in flow, but leaving it for now in the sign-up fixing it in this issue #94